.github/workflows/demo-strapi-version-check.yml

```diff
@@ -0,0 +1,96 @@
+name: 'Demo App Strapi Version Check'
+
+on:
+  schedule:
+    - cron: '0 23 * * 3' # Run weekly on Wednesday at 23:00 UTC
+  workflow_dispatch: # Allow manual trigger
+
+permissions:
+  contents: write # to create branch and PR
+  pull-requests: write # to create PR
+
+jobs:
+  check-and-update:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9.1.0
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Get current version from package.json
+        id: current-version
+        run: |
+          CURRENT_VERSION=$(node -p "require('./demo/.strapi-app/package.json').dependencies['@strapi/strapi']")
```

Comment:
With exact version being listed in `package.json`, it allows the action to use it as state and derive the comparison from it.
If we go back to latest, we'd have to either:
- `pnpm install` then `pnpm up ...@latest` then, look into the `node_modules/@strapi/strapi/package.json` to observe a diff, then conditionally make the PR.
- Rely on reading the lock file with a regex to match the @strapi/strapi version declaration, which I find clunky

@Convly @innerdvations can you detail the rationale/preference behind keeping `latest` instead of a pinned version?